### PR TITLE
Automated Minimum Dependency Updates

### PR DESCRIPTION
**Pull request by machineFL**
This is an auto-generated PR with **minimum** dependency updates. Please do not delete the `min-dep-update` branch because it's needed by the auto-dependency bot.

### CODE REVIEW

**Comment by dsherry** [comment]
@gsheni this is because you turned on the minimum dependency checker in evalml, right?

**Comment by gsheni** [comment]
@dsherry yes, this was generated due to workflow I turned on in this branch:
- https://github.com/alteryx/evalml/pull/2267/
- Once #2267 is merged in, we can merge in this PR. That PR also has a fix to skip the release notes check on min-deps branches.

**Review by dsherry — APPROVED** [review]
Looks good. Let's merge this after #2267 since that PR adds the GH action which generated this.

**Pull request merged**
